util/reggen/ip_block.py

```diff
@@ -291,6 +317,125 @@ def from_path(path: str,
+        # Alias register definitions are only compatible with named devices.
+        if alias_bus_interfaces.has_unnamed_device or\
+           self.bus_interfaces.has_unnamed_device:
+            raise ValueError("Alias registers must use named devices "
+                             "(in block at {}).".format(where))
+
+        # Check that the device interface names are
+        # a subset of the already defined register blocks
+        bus_device_names = set(self.bus_interfaces.named_devices)
+        alias_bus_device_names = set(alias_bus_interfaces.named_devices)
+        if not alias_bus_device_names.issubset(bus_device_names):
+            raise ValueError("Alias file {} refers to device names {} that "
+                             "do not map to device names in {}."
+                             .format(where, list(alias_bus_device_names),
+                                     self.name))
+
+        self.alias_impl = check_name(rd['alias_impl'],
+                                     'alias_impl of block at ' + where)
+
+        alias_target = check_name(rd['alias_target'],
+                                  'alias_target of block at ' + where)
+
+        if alias_target != self.name:
+            raise ValueError("Alias target block name {} in {} "
+                             "does not match block name {}."
+                             .format(alias_target, where, self.name))
+
+        init_block = RegBlock(self.regwidth, self.params)
+
+        alias_reg_blocks = RegBlock.build_blocks(init_block, rd['registers'],
+                                                 self.bus_interfaces,
+                                                 self.clocking,
+                                                 True)
+
+        # Check that alias register block names are
+        # a subset of the already defined register blocks
+        alias_reg_block_names = set(alias_reg_blocks.keys())
+
+        if not alias_reg_block_names.issubset(set(self.reg_blocks.keys())):
+            raise ValueError("Alias file {} refers to register blocks {} that "
+                             "do not map to register blocks in {}."
+                             .format(where, list(alias_reg_block_names),
+                                     self.name))
+
+        # Check that the alias bus interface names and register blocks match
+        if alias_reg_block_names != alias_bus_device_names:
+            raise ValueError("Interface and register block names do not match "
+                             "in {}.".format(where))
+
+        # Validate alias registers against the generic reg blocks,
+        # and enhance the information in the existing datastructures.
+        for block_key, alias_block in alias_reg_blocks.items():
+            # Double check the interface definition options
+            if self.bus_interfaces.device_async:
+                if not alias_bus_interfaces.device_async:
+                    raise ValueError('Inconsistent configuration of '
```

Comment:
Nit: I know it's painful, but maybe better to tweak this error message and the one below so that they are distinct (making debugging of a failed check a bit easier)

Reply:
Jup, will make these messages unique.